xcode/Subconscious/Shared/Services/Noosphere/Sphere.swift

```diff
@@ -16,56 +16,71 @@ public protocol SphereProtocol {
     
     func version() throws -> String
     
-    func getFileVersion(slashlink: String) -> String?
+    func getFileVersion(slashlink: Slashlink) -> String?
     
     func readHeaderValueFirst(
-        slashlink: String,
+        slashlink: Slashlink,
         name: String
     ) -> String?
     
-    func readHeaderNames(slashlink: String) -> [String]
+    func readHeaderNames(slashlink: Slashlink) -> [String]
     
-    func read(slashlink: String) throws -> Memo
+    func read(slashlink: Slashlink) throws -> Memo
     
     func write(
-        slug: String,
+        slug: Slug,
         contentType: String,
         additionalHeaders: [Header],
         body: Data
     ) throws
     
-    func remove(slug: String) throws
+    func remove(slug: Slug) throws
     
     func save() throws -> String
     
-    func list() throws -> [String]
+    func list() throws -> [Slug]
     
     func sync() throws -> String
     
-    func changes(_ since: String?) throws -> [String]
+    func changes(_ since: String?) throws -> [Slug]
+    
+    func getPetname(petname: Petname) throws -> String
+    
+    func setPetname(did: String, petname: Petname) throws
+    
+    func unsetPetname(petname: Petname) throws
+    
+    func resolvePetname(petname: Petname) throws -> String
+    
+    func listPetnames() throws -> [Petname]
+    
+    func getPetnameChanges(sinceCid: String) throws -> [Petname]
```

Comment:
I assume petname changes are an array of petnames that changed. Can you confirm @bfollington?

Reply:
Yep, afaik it's a list of plain petnames that have been added, removed or had the identity changes (with nothing to distinguish between those cases in the result).